x/ccv/provider/keeper/hooks.go

```diff
@@ -48,13 +48,14 @@ func (h Hooks) AfterUnbondingInitiated(ctx sdk.Context, ID uint64) {
 
 	// Set unbondingOp
 	if err := h.k.SetUnbondingOp(ctx, unbondingOp); err != nil {
-		panic(fmt.Errorf("unbonding op could not be persisted: %w", err))
+		return fmt.Errorf("unbonding op could not be persisted: %w", err)
```

Comment:
What happens on the other side when this error is returned? It's hard to find out what actual branch you're using from the go mod file.

Reply:
I believe a logged error: https://github.com/cosmos/cosmos-sdk/blob/9c145c827001222df2e3e1101010874aeac20997/x/staking/keeper/delegation.go#L527

Reply:
Previously we panicked for this error, and now we do not. Is this change in behavior ok and intentional? Does it even matter?

Reply:
@jtremback  the SDK branch used is in the description

@smarshall-spitzbart `RunTx` normally recovers all the panic in tx executions https://docs.cosmos.network/v0.45/core/runtx_middleware.html so I think doesn't change the logic and it's more a matter of logging the error.

Reply:
I agree that this should return an error instead of panicking, especially since `AfterUnbondingInitiated` has error as a return field

Reply:
> I agree that this should return an error instead of panicking, especially since AfterUnbondingInitiated has error as a return field.

The error return field is introduced in this PR, I'm just pointing out that I think this PR will change behavior. Whether that matters, I'm not sure.

From the link @sainoe has given
> Depending on the panic type a different handler is used, for instance the default one prints an error log message.

I agree that we will log an error before and after this PR👍 

There is still a change in behavior AFAIK tho. If we panic, execution of the tx will halt and not be committed. Whereas with the new functionality, we log an error and continue tx execution. 

To make this PR not change any functionality, we'd need to panic on the sdk side of things

Reply:
Yea we need to leave the panics. They roll back the tx